flow/connectors/bigquery/merge_stmt_generator.go

```diff
@@ -231,9 +226,9 @@ func (m *mergeStmtGenerator) generateUpdateStatements(
 		// generates update statements for the case where updates and deletes happen in the same branch
 		// the backfill has happened from the pull side already, so treat the DeleteRecord as an update
 		// and then set soft-delete to true.
-		if peerdbCols.SoftDelete && (peerdbCols.SoftDeleteColName != "") {
+		if m.peerdbCols.SoftDelete && (m.peerdbCols.SoftDeleteColName != "") {
```

Comment:
Calculate this bool at start, if it's true `updateStmts` should allocate twice as much capacity